components/brave_mobile_subscription/renderer/android/subscription_render_frame_observer.cc

```diff
@@ -268,11 +283,25 @@ bool SubscriptionRenderFrameObserver::IsAllowed() {
     product_ = Product::kVPN;
   } else if (product == kProductLeoParamValue) {
     product_ = Product::kLeo;
+    // We allow to inject linkResult object if intent value is empty and path is
+    // /order-link/ as https://account.brave.com?intent=link-order&product=leo
+    // gets redirected to https://account.brave.com/order-link/?product=leo for
+    // an actual linking where we should receive the result of linking
+    if (intent.empty()) {
+      std::string path = ExtractPath(current_url);
```

Comment:
prefer std::string_piece (and return url.path_piece() from ExtractPath). Frankly i wouldn't even write `ExtractPath` b/c when i read this code i assume that the function does something non-trivial